### DISCUSSION TITLE
Proposal automated exports for xbvr.

### DISCUSSION THREAD

**Discussion by Tweeticoats** [stashapp/stash-box#355] [Ideas]
### Description:
XBVR is an open source video manager for VR porn see https://github.com/xbapps/xbvr
This has it's own set of scrapers written in go and will automatically run these scrapers and match them to file names.
As these are compiled in to the binary they cannot be easily changed and break form time to time requiring a new release of the program.
XBVR does have a json export / import format for importing bundles of scenes.
As we are adding metadata of all video including many VR studios I would like to start generating these json bundles from stashdb and make them available to xbvr.
There are some studios that no longer exist and some studios occasionally remove scenes so a place to manually create these scenes would be useful.

I have started writing the script as a proof of concept to do this but it needs further testing and work before it is ready for production.
https://github.com/Tweeticoats/stashdb2xbvr

### Needed resources:
- I would a place to store these exports for example https://xbvr.stashdb.org or http://exports.stashdb.org
- I need a little storage to store these files as the bundles are small this should only need to be a few mb of files.
- This should have a file generated for each VR studio as well as an export of each studio or network with a consistent url so this url can be shared and possibly embedded in to xbvr.
- It would be useful to have this script run as a cron job automatically every few hours.

### Example json structure:
{
    "timestamp": "2030-01-01T00:00:00.6984107Z",
    "bundleVersion": "1",
    "scenes": [
        {
        "_id": "vrhush-xxxxx",
        "scene_id": "xxxx",
        "scene_type": "VR",
        "title": "Title of Scene",
        "studio": "VRHush",
        "site": "VRHush",
        "covers": [
            "url"
        ],
        "gallery": [
            "url"
        ],
        "tags": ["tag1"],
        "cast": [
            "performer"
        ],
        "filename": [
            "VRHush_xxxxxxx_1k.mp4",
            "VRHush_xxxxxxx_2k.mp4",
            "VRHush_xxxxxxx_4k.mp4",
            "VRHush_xxxxxxx_6k.mp4",
            "VRHush_xxxxxxx_8k.mp4"
        ],
        "synopsis": "description",
        "duration": 0,
        "released": "2021-12-05",
        "homepage_url": "url"
        }
    ]
}

**Comment by ghost** [comment]
Wouldn't it be better to have a stash-box scraper in xbvr that could query files, or is that the purpose of this export?

**Reply by Tweeticoats** [reply]
Having us generate something for them or having them fully integrate with stashdb are both valid options.

The advantage of having us generate exports of metadata is the software already has a feature to import json files over http.

The xbvr project does not have many developers so feel free to help out on their side it is written in go in a similar way to stash.
XBVR matches based on file names and this works ok as the file name patterns are predictable.
I have created an issue and mentioned it in discord for them to look in to.
